### PR TITLE
feature(eos_designs): Add support for A/A model to connect L2LEAF

### LINKED ISSUE #807: Support ESI AA port-channel to connect L2LEAF uplinks
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**
Currently, eos_designs role assumes that L2LEAFs are connected to L3LEAF MLAG pair.
This feature request is to add configuration knob to choose which type of multihoming to use for L2LEAF.

**Describe the solution you'd like**
Currently, eos_designs role assumes that L2LEAFs are connected to L3LEAF MLAG pair.
"short_esi" value for L2LEAF should have some default values based on the Port-Channel number and pair id.

**Describe alternatives you've considered**

define exact "short_esi" per L2LEAF node_group

**Additional context**

### PR DESCRIPTION
**Pull request by titom73**
## Change Summary
EVPN A/A to connect L2LEAF to L3LEAF

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes #807

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Add support of EVPN A/A to attach L2LEAF to L3LEAF nodes.

On L2LEAF nodes, the minimum data model is:

```yaml
l2leaf:
  node_groups:
    DC1_L2LEAF1:
      short_esi: 0808:0707:0606
      parent_l3leafs: [ DC1-LEAF2A, DC1-LEAF2B ]
      nodes:
        DC1-L2LEAF1A:
          l3leaf_interfaces: [ Ethernet7, Ethernet7 ]
```

> The scenario is limited to dual attachment of an L2LEAF to 2 L3LEAFs in the same node groups. Does not support connectivity redundancy from a SERVER standpoint.

## How to test

Check molecule configuration: https://github.com/titom73/ansible-avd/blob/1591872bca2b4bc74339130dc36c403ea68469cc/ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/documentation/devices/DC1-LEAF2B.md [hunk 1]
```diff
@@ -483,7 +483,12 @@ interface Port-Channel7
    switchport
    switchport trunk allowed vlan 110-111,120-121,130-131,160-161
    switchport mode trunk
-   mlag 7
+   !
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/documentation/devices/DC1-LEAF2B.md]: this `!` should not be there. It is just another config line within the same section of the interface. A show running-config would not have this either.

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md [hunk 2]
```diff
@@ -422,8 +422,40 @@ l2leaf:
...
+      parent_l3leafs: [ DC1-SVC3A ]
+      mlag: false
+      nodes:
+
+        # First node.
+        < l2_leaf_inventory_hostname_2 >:
+          id: < integer >
+          mgmt_ip: < IPv4_address/Mask >
+          l3leaf_interfaces: [ < ethernet_interface_7 >, < ethernet_interface_7 > ]
+          short_esi: < short esi value >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md]: I would think this should be under the node_group and not the individual node. If we have l2leafs in MLAG, they would share the same ESI on the upstream l3leafs.

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md [hunk 3]
```diff
@@ -422,8 +422,40 @@ l2leaf:
           id: < integer >
           mgmt_ip: < IPv4_address/Mask >
           l3leaf_interfaces: [ < ethernet_interface_8 >, < ethernet_interface_8 > ]
+
+    # node_group_3, will result in Active/Active connection to L3LEAFs.
+    < node_group_3 >:
+      parent_l3leafs: [ DC1-SVC3A ]
+      mlag: false
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md]: This `mlag: false` is not needed. Actually we might run MLAG on the l2leaf and A/A on the l3leaf, so this is misleading.

**Review by ClausHolbechArista — DISMISSED** [review]

File: ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml [hunk 4]
```diff
@@ -148,6 +148,7 @@ l2leaf:
   node_groups:
     DC1_L2LEAF1:
       mlag: false
+      esi: true
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml]: Why is this needed? Could we not just detect the `short_esi`? Espcially if we move it to `node_group`. Thanks

File: ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/documentation/devices/DC1-LEAF2B.md (hunk 1 above)
Comment (titom73) [reply ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/documentation/devices/DC1-LEAF2B.md]: Resolved in last commit (d3852ae)

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md (hunk 2 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md]: Moved at node_group level with commit e3c7310

File: ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md (hunk 3 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_designs/doc/l3ls-evpn/fabric-topology.md]: Not required anymore with commit e3c7310

File: ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml (hunk 4 above)
Comment (titom73) [reply ansible_collections/arista/avd/molecule/evpn_underlay_ebgp_overlay_ebgp/inventory/group_vars/DC1_FABRIC.yml]: Removed as per commit e3c7310

**Review by ClausHolbechArista — APPROVED** [review]
LGTM!

**Comment by ClausHolbechArista** [comment]
Moved to #878
